### ISSUE TITLE
'limit' seems to not consider query criteria

### ISSUE THREAD

**Issue by tolbertam** [tolbertam/sstable-tools#36]
It looks like the limit support does not consider query criteria, i.e.:

```
cqlsh> select * from sstable where ticker='YHOO' limit 5;
 ┌─────────┬─────────────────────┬────────────┬─────────┬─────────┬─────────┬─────────┬─────────┐
 │ticker   │date                 │adj_close   │close    │high     │low      │open     │volume   │
 ╞═════════╪═════════════════════╪════════════╪═════════╪═════════╪═════════╪═════════╪═════════╡
 │ORCL     │2016-02-19 00:00-0600│36.779999   │36.779999│36.790001│36.419998│36.52    │13118400 │
 ├─────────┼─────────────────────┼────────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
 │ORCL     │2016-02-18 00:00-0600│36.630001   │36.630001│36.869999│36.400002│36.709999│12464800 │
 ├─────────┼─────────────────────┼────────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
 │ORCL     │2016-02-17 00:00-0600│36.630001   │36.630001│36.77    │35.970001│35.970001│13146600 │
 ├─────────┼─────────────────────┼────────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
 │ORCL     │2016-02-16 00:00-0600│35.700001   │35.700001│35.91    │35.419998│35.759998│18685400 │
 ├─────────┼─────────────────────┼────────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
 │ORCL     │2016-02-12 00:00-0600│35.540001   │35.540001│35.549999│34.91    │35.240002│15806800 │
 └─────────┴─────────────────────┴────────────┴─────────┴─────────┴─────────┴─────────┴─────────┘
```

**Comment by tolbertam** [comment]
Easier than I anticipated, fixed by 430c942

**Comment by tolbertam** [comment]
This isn't a complete fix because it only takes in account the primary key range and doesn't include the non PK restrictions.  I can't tell if there is a way to solve this easily (there may not be).
